js/flamegraph.js

```diff
@@ -159,42 +159,28 @@ function drawFlameGraph(function_calls) {
     .attr('height', (d, _i) => d.height)
     .attr('label', (d, _i) => d.name)
     .style('fill', (d, _i) => d.colour)
+    .style('stroke', 'rgb(0,0,0)')
     .on('click', function(d, _i) {
-      // De-select the current selection and select the new one.
-      if (currentSelection) {
-        d3.select(currentSelection)
-        .style('stroke-width', 0)
-        .style('stroke', 'rgb(0,0,0)');
-      }
-      d3.select(this)
-        .style('stroke-width', 3)
-        .style('stroke', 'rgb(0,0,0)');
-      currentSelection = this;
+      currentSelection = d;
       selectNode(d);
+      highlightSelectedNode();
     })
     .append('title').text((d, _i) => createStack(d));
 
-  // Add text labels to each box.
-  // let textSelection = svg.selectAll("text")
-  //   .data(function_calls);
-  //
-  // // Only update x and width on update, labels never change.
-  // textSelection
-  //   .attr("x", (d, i) => {if( isNaN(d.x_pos) ) { console.error("NaN x_pos for " + d.name) } else { return d.x_pos}})
-  //   .attr("width", (d, i) => {if( isNaN(d.width) ) { console.error("NaN width for " + d.name) } else {return d.width}})
-  //
-  // textSelection
-  //   .enter()
-  //   .append("text")
-  //   .attr("x", (d, i) => {if( isNaN(d.x_pos) ) { console.error("NaN x_pos for " + d.name) } else { return d.x_pos}})
-  //   .attr("y", (d, i) => (flamegraphCanvasHeight - (d.height/2)) - d.y_pos)
-  //   .attr("width", (d, i) => {if( isNaN(d.width) ) { console.error("NaN width for " + d.name) } else {return d.width}})
-  //   .attr("height", (d, i) => d.height)
-  //   .text((d, i)=> d.name)
-  //   .attr("fill", "black");
-  //   // .style("stroke-width", 3)
-  //   // .style("stroke", "rgb(0,0,0)");
+  highlightSelectedNode();
+}
+
 
+function highlightSelectedNode() {
+  // Set the highlighting and make the selected element the last to be
+  // drawn to stop the border being covered on some edges by other rectangles.
+  d3.selectAll('rect')
```

Comment:
This selects every single `rect` in the dashboard, including the ones making up the graph titles - which wipes out both the titles and the maximise buttons. I think you wanted to restrict this to the 'flamegraph' SVG element.